test/Hintman/Suggest.hs

```diff
@@ -0,0 +1,57 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+module Hintman.Suggest (tests) where
+
+import Hedgehog
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+import Relude
+import qualified Data.Text as T
+
+data LineState = Plain | Trailing
+
+space :: MonadGen m => m Char
+space = pure ' '
+
+genSpaces :: Int -> Int -> Gen Text
+genSpaces i j = Gen.text (Range.linear i j) space
+
+genPlain :: Gen Text
+genPlain =
+  T.intercalate " " <$>
+  Gen.list (Range.linear 0 20) (
+      Gen.text (Range.linear 5 100) Gen.alpha)
+
+genLines :: LineState -> Gen Text
+genLines Plain
+  = T.intercalate "\n" <$>
+    Gen.list (Range.linear 0 120) genPlain
+genLines Trailing
+  = T.intercalate "\n" <$>
+    Gen.list (Range.linear 0 120) genTrailing
+  where
+    genTrailing :: Gen Text
+    genTrailing
+      = Gen.subterm2 genPlain (genSpaces 1 20) T.append
+
+{-
+prop_nothing_if_no_trailing_spaces :: Property
+prop_nothing_if_no_trailing_spaces = undefined
+
+prop_suggest_when_trailing_spaces :: Property
+prop_suggest_when_trailing_spaces = undefined
+
+prop_suggest_suggest_should_be_nothing :: Property
+prop_suggest_suggest_should_be_nothing = undefined
+-}
+
+prop_placeholder :: Property
+prop_placeholder =
+  property $ do
+    xs <- forAll $ Gen.list (Range.linear 0 100) Gen.alpha
+    reverse (reverse xs) === xs
+
+tests :: IO Bool
+tests = checkParallel $$(discover)
```

Comment:
Instead of `TemplateHaskell` and $$ discover it's better to create `Group` manually and name tests in some format. This approach has two pros:

1. No need to name property tests in weird style with `prop` prefix and underscores inside
2. You can assign more meaningful names to the tests.